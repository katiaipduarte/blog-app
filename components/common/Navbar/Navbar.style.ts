import styled from 'styled-components';

export const NavbarContainer = styled.nav`
  .menu {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    list-style-type: none;

    ul {
      li a {
        display: block;
        padding: 1.5rem 0.5rem;
        text-decoration: none;
      }
    }

    .toggle {
      order: 1;
      a {
        svg {
          height: 2rem;
        }
      }
    }

    .logo {
      font-size: 2rem;
      padding: 0.75rem 1rem 0.75rem 0;
      color: ${({ theme }) => theme.primaryColor};
    }

    .item {
      padding: 1rem;
      order: 3;
      width: 100%;
      text-align: center;
      display: none;
      a:hover,
      a:hover::after {
        color: ${({ theme }) => theme.secondaryColor};
      }
      font-size: 1.25rem;
      border-top: 1px solid ${({ theme }) => theme.text};
      border-bottom: 1px solid ${({ theme }) => theme.text};
    }
  }
  .active .item {
    display: block;
  }

  .theme-toggle {
    display: none;
  }

  /* Tablet menu */
  @media all and (min-width: 700px) {
    .menu {
      justify-content: center;
      .logo {
        flex: 1;
      }
      .toggle {
        flex: 1;
        text-align: right;
        order: 2;
      }
    }
  }

  /* Desktop menu */
  @media all and (min-width: 960px) {
    .menu {
      align-items: flex-start;
      flex-wrap: nowrap;
      background: none;
      .logo {
        order: 0;
      }
      .item {
        order: 1;
        position: relative;
        display: block;
        width: auto;
        font-size: 1rem;
        border: none;
      }
      .toggle {
        display: none;
      }
      .theme-toggle {
        order: 2;
        display: inherit;
        padding: 1rem;
      }
    }
  }
`;
