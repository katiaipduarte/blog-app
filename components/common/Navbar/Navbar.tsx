import React, { useState } from 'react';
import Link from 'next/link';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import ThemeSwitcher from '@components/ui/ThemeSwitcher/ThemeSwitcher';
import { NavbarContainer } from './Navbar.style';

type Props = {
  mountedComponent: boolean;
  themeToggler: () => void;
  theme: string;
};

const Navbar = (props: Props): JSX.Element => {
  const { mountedComponent, themeToggler, theme } = props;

  const [seeMobileMenu, setSeeMobileMenu] = useState<boolean>(false);
  if (!mountedComponent) return <div />;

  return (
    <NavbarContainer>
      <ul className={seeMobileMenu ? 'menu active' : 'menu'}>
        <li className="logo">
          <Link href="/">BlogApp</Link>
        </li>
        <li className="item">
          <Link href="/blog">Blog</Link>
        </li>
        <li className="theme-toggle">
          <ThemeSwitcher theme={theme} toggleTheme={themeToggler} />
        </li>
        <li className="toggle">
          <ThemeSwitcher theme={theme} toggleTheme={themeToggler} />
          <a
            role="button"
            tabIndex={0}
            aria-label="Click to open mobile menu"
            onClick={() => setSeeMobileMenu(!seeMobileMenu)}
          >
            <FontAwesomeIcon icon={faBars} />
          </a>
        </li>
      </ul>
    </NavbarContainer>
  );
};

export default Navbar;
