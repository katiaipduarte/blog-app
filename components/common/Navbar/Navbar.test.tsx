import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Navbar from './Navbar';

describe('<Navbar /> component Unit Test', () => {
  beforeEach(() => {
    render(
      <Navbar
        mountedComponent={true}
        theme="light"
        themeToggler={jest.fn().mockName('mockedFunction')}
      />
    );
  });

  it('should render 1 <Navbar /> component', () => {
    const { container } = render(
      <Navbar
        mountedComponent={true}
        theme="light"
        themeToggler={jest.fn().mockName('mockedFunction')}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add active to className in ul, when clicking the faBars icon', () => {
    const menuBtn = screen.getByTestId('menu-btn');
    const menu = screen.getByTestId('menu');

    expect(menuBtn).toBeTruthy();
    expect(menu.className).toEqual('menu');

    fireEvent.click(screen.getByTestId('mobile-menu-btn'));
    expect(menu.className).toEqual('menu active');
  });
});
