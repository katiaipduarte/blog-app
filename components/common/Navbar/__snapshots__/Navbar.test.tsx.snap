// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> component Unit Test should render 1 <Navbar /> component 1`] = `
<nav
  class="Navbarstyle__NavbarContainer-sc-151odup-0 ikyQcA"
>
  <ul
    class="menu"
    data-testid="menu"
  >
    <li
      class="logo"
    >
      <a
        href="/"
      >
        BlogApp
      </a>
    </li>
    <li
      class="item"
    >
      <a
        href="/blog"
      >
        Blog
      </a>
    </li>
    <li
      class="theme-toggle"
    >
      <button
        aria-label="Toggle dark mode"
        class="ThemeSwitcherstyle__ThemeSwitcherButton-sc-1ygfhkk-0 WaSyQ"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-moon fa-w-16 "
          data-icon="moon"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
            fill="currentColor"
          />
        </svg>
      </button>
    </li>
    <li
      class="toggle"
      data-testid="menu-btn"
    >
      <button
        aria-label="Toggle dark mode"
        class="ThemeSwitcherstyle__ThemeSwitcherButton-sc-1ygfhkk-0 WaSyQ"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-moon fa-w-16 "
          data-icon="moon"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
            fill="currentColor"
          />
        </svg>
      </button>
      <a
        aria-label="Click to open mobile menu"
        data-testid="mobile-menu-btn"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-w-14 "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;
